src/jsx/components/results_page.jsx

```diff
@@ -38,10 +40,14 @@ class ResultsPage extends React.Component {
     // Decide if data is a URL or the results JSON
     var newData;
     if (typeof this.props.data == "string") {
-      d3.json(this.props.data, function(data) {
-        //self.setState({ json: data });
-        newData = data;
-      });
+      if (this.props.data != self.state.jsonPath) {
+        d3.json(this.props.data, function(data) {
+          //self.setState({ json: data });
+          newData = data;
+        });
+      } else {
+        newData = this.state.json;

```

Comment:
Totally trivial comment, but could probably save a few lines by setting newData to this.state.json on line 41. That way you wouldn't need an else statement here.

Reply:
Good call. Trivial as it may be, this kind of thing is always helpful. Thanks!